frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/toolbar/DiscoverAction.java

```diff
@@ -127,7 +130,17 @@ private void saveDiscoveredLayout(IGitMacheteRepositorySnapshot repositorySnapsh
       @Override
       @SneakyThrows
       public void run(ProgressIndicator indicator) {
-        branchLayoutWriter.write(macheteFilePath, branchLayout, /* backupOldLayout */ true);
+        ApplicationManager.getApplication()
+            .invokeLater(() -> WriteAction
+                .compute(() -> {
+                  try {
+                    branchLayoutWriter.write(macheteFilePath, branchLayout, /* backupOldLayout */ true);
+                  } catch (BranchLayoutException e) {
+                    throw new RuntimeException(e);
+                  }
+                  return true;
```

Comment:
return true just to return smth, right? if so, we probably could use 
`com.intellij.openapi.application.WriteAction#run(com.intellij.util.ThrowableRunnable<E>)` here